File: backend/src/main/java/zw/co/hisolutions/core/service/DistributionMethodService.java
package zw.co.hisolutions.core.service;
 
import zw.co.hisolutions.core.entity.DistributionMethod;
import zw.co.hisolutions.common.service.GenericService;

public interface DistributionMethodService extends GenericService<DistributionMethod, Long>{
    public DistributionMethod findByName(String name);   
}

File: frontend/src/app/admin/services/rest/distribution-method.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {DataService} from 'shared/services/data.service';

@Injectable()
export class DistributionMethodService extends DataService  {

      constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/distributionmethods'; 
        super(httpClient, resourcePath);
    }

}
